tst/CTA.WebForms2Blazor.Tests/FileConverters/ViewFileConverterTests.cs

```diff
@@ -275,5 +267,155 @@ public void ConvertEmbeddedCode_Converts_SingExpr()
+        </Columns>
+    </GridView>
+</div>
+<div>
+    <GridView AutoGenerateColumns=""false"" DataKeyNames=""CustomerID"" SelectMethod=""GetCustomers"" EmptyDataText=""No data available"" ItemType=""PleaseReplaceWithActualItemTypeHere"">
+        <Columns>
+            <BoundField DataField=""CustomerID"" HeaderText=""ID"" ItemType=""PleaseReplaceWithActualItemTypeHere""></BoundField>
+            <BoundField DataField=""CompanyName"" HeaderText=""CompanyName"" ItemType=""PleaseReplaceWithActualItemTypeHere""></BoundField>
+            <BoundField DataField=""FirstName"" HeaderText=""FirstName"" ItemType=""PleaseReplaceWithActualItemTypeHere""></BoundField>
+            <BoundField DataField=""LastName"" HeaderText=""LastName"" ItemType=""PleaseReplaceWithActualItemTypeHere""></BoundField>
+            <TemplateField ItemType=""PleaseReplaceWithActualItemTypeHere"">
+                <ItemTemplate Context=""Item"">
+                    <button type=""button"">Click Me! @(Item.Name )</button>
+                </ItemTemplate>
+            </TemplateField>
+            <ButtonField ButtonType=""Button"" DataTextField=""CompanyName"" DataTextFormatString=""{0}"" CommandName=""Customer"" ItemType=""PleaseReplaceWithActualItemTypeHere""></ButtonField>
+        </Columns>
+    </GridView>
+</div>";
+            
+            Assert.AreEqual(expectedContents, fileContents);
+            Assert.True(fileContents.Contains("GridView"));
+            Assert.True(fileContents.Contains(@"ref=""GridView1"""));
+            Assert.False(fileContents.Contains("asp:GridView"));
+            Assert.False(fileContents.Contains("asp:BoundField"));
+            Assert.AreEqual(fi.RelativePath, relativePath);
+        }
+
+        [Test]
+        public void TestUpdateInnerHtmlNode_Matches_Id()
+        {
+            var htmlString = @"<div class=""esh-table"">
+    <asp:ListView ID=""productList"" ItemPlaceholderID=""itemPlaceHolder"" runat=""server"" ItemType=""eShopLegacyWebForms.Models.CatalogItem"">
+        <LayoutTemplate>
+            <table class=""table"">
+                <tbody>
+                    <asp:PlaceHolder runat=""server"" ID=""itemPlaceHolder""></asp:PlaceHolder>
+                </tbody>
+            </table>
+        </LayoutTemplate>
+    </asp:ListView>
+</div>";
+            var htmlNode = HtmlNode.CreateNode(htmlString);
+            var testControlConverter = new ListViewControlConverter();
+            var correctNameAndId = testControlConverter.UpdateInnerHtmlNode(htmlNode, "asp:PlaceHolder", "itemPlaceHolder");
+            var correctNameAndNoId = testControlConverter.UpdateInnerHtmlNode(htmlNode, "asp:PlaceHolder");
+            var correctNameAndIncorrectId = testControlConverter.UpdateInnerHtmlNode(htmlNode, "asp:PlaceHolder", "inccorectid");
+            
+            Assert.True(correctNameAndId);
+            Assert.True(correctNameAndNoId);
+            Assert.False(correctNameAndIncorrectId);
+        }
```

Comment:
Is the input node of `UpdateInnerHtmlNode()` mutated? If so, we should verify correctness of that behavior as well. I would suggest breaking this into 3 separate unit tests called:
* `UpdateInnerHtmlNode_Succeeds_On_ElementName_And_Id_Match`
* `UpdateInnerHtmlNode_Succeeds_On_Id_Match`
* `UpdateInnerHtmlNode_Fails_On_No_Id_Match`

In each test, Assert True/False if the function succeeds (like you currently have), but also assert that some expected string literal matches the string form of the modified htmlNode.

Reply:
Well I was planning for this unit test to only make sure that the Id matching works properly, and test for overall correctness in other unit tests. Is that a good idea separate tests this way or is it necessary to split the test up further and verify overall correctness everytime?